### ISSUE TITLE
CA1508 False positive on declaration using statement

### ISSUE THREAD

**Issue by julealgon** [dotnet/roslyn-analyzers#4526]
### Analyzer

**Diagnostic ID**: [CA1508 ](https://docs.microsoft.com/dotnet/fundamentals/code-analysis/quality-rules/CA1508): `Avoid dead conditional code`

### Analyzer source

**SDK**: [Built-in CA analyzers in .NET 5 SDK or later](https://docs.microsoft.com/dotnet/fundamentals/productivity/code-analysis)

**Version**: [SDK 5.0.100](https://dotnet.microsoft.com/download/dotnet/5.0)

### Describe the bug

CA1508 is firing when creating a new instance inside a `using` statement without declaring a variable.

### Steps To Reproduce

1. Create `using` block with new `IDisposable` object

```csharp
    // Assert
    using (new AssertionScope())
    {
        ...
    }
```

2. Observe false positive warning:
> 'new AssertionScope()' is never 'null'. Remove or refactor the condition(s) to avoid dead code.

![image](https://user-images.githubusercontent.com/461579/101408154-c54a8580-38ba-11eb-8013-af80d57d699a.png)

### Expected behavior

No warning should be raised here.

### Actual behavior

Warning is raised.

## Additional context

This is a more specific version of #2985

The warning text talks about a "condition", even though there are no conditional statements in the offending code. Why is it even being considered on a `using` block if it is related to conditional code?

**Comment by mavasani** [comment]
@julealgon Can you try https://www.nuget.org/packages/Microsoft.CodeAnalysis.NetAnalyzers/5.0.1? I believe this was recently fixed, but not yet shipped with .NET SDK yet.

**Comment by julealgon** [comment]
> @julealgon Can you try https://www.nuget.org/packages/Microsoft.CodeAnalysis.NetAnalyzers/5.0.1? I believe this was recently fixed, but not yet shipped with .NET SDK yet.

Thanks @mavasani . Yes, I can confirm that referencing the newer package fixes the problem for me. Closing.

What exactly is the recommendation here from Microsoft? Is it preferred to manually reference `Microsoft.CodeAnalysis.NetAnalyzers` or is it better to rely on the SDK version?

If SDK updates are not frequent enough, I wonder if manually referencing is the better option. If so, I'd probably do it via directory.build.props file.
